frontend/appflowy_tauri/src/appflowy_app/components/_shared/database-hooks/useCell.ts

```diff
@@ -0,0 +1,40 @@
+import { CellIdentifier } from '../../../stores/effects/database/cell/cell_bd_svc';
+import { CellCache } from '../../../stores/effects/database/cell/cell_cache';
+import { FieldController } from '../../../stores/effects/database/field/field_controller';
+import { CellControllerBuilder } from '../../../stores/effects/database/cell/controller_builder';
+import { DateCellDataPB, FieldType, SelectOptionCellDataPB } from '../../../../services/backend';
+import { useState } from 'react';
+
+export const useCell = (cellIdentifier: CellIdentifier, cellCache: CellCache, fieldController: FieldController) => {
+  const [data, setData] = useState<string[]>([]);
+
+  const loadCell = async () => {
+    const builder = new CellControllerBuilder(cellIdentifier, cellCache, fieldController);
+    const cellController = builder.build();
+    cellController.subscribeChanged({
+      onCellChanged: (value) => {
+        if (
+          cellIdentifier.fieldType === FieldType.Checklist ||
+          cellIdentifier.fieldType === FieldType.MultiSelect ||
+          cellIdentifier.fieldType === FieldType.SingleSelect
+        ) {
+          const v = value.unwrap() as SelectOptionCellDataPB;
+          setData(v.select_options.map((option) => option.id));
+        } else if (cellIdentifier.fieldType === FieldType.DateTime) {
+          const v = value.unwrap() as DateCellDataPB;
+          setData([v.date]);
+        } else {
+          const v = value.unwrap() as string;
+          setData([v]);
+        }
+      },
+    });
+
+    cellController.getCellData();
```

Comment:
<img width="593" alt="image" src="https://user-images.githubusercontent.com/86001920/222898416-b1f79abe-bfaf-47a7-9aa4-2b393ad943f0.png">
Do you see this warning in your IDE?

Reply:
fixed